server/pulp/server/webservices/middleware/exception.py

```diff
@@ -19,65 +6,20 @@
 
 from django.http import HttpResponse, HttpResponseServerError
 
-from pulp.server.compat import json, http_responses
+from pulp.server.compat import json
 from pulp.server.exceptions import PulpException
-from pulp.server.webservices import serialization
+from pulp.server.webservices.views import serializers
 
 
 logger = logging.getLogger(__name__)
 
 
-class ExceptionHandlerMiddleware(object):
-    """
-    Catch otherwise unhandled exceptions and return appropriate 500 responses.
-    @ivar app: WSGI application or middleware
-    @ivar debug: boolean flag when true, puts the traceback into the response
-    """
-
-    def __init__(self, app, debug=False):
-        self.app = app
-        self.debug = debug
-        self.headers = {'Content-Encoding': 'utf-8',
-                        'Content-Type': 'application/json',
-                        'Content-Length': '-1'}
-
-    def __call__(self, environ, start_response):
-        try:
-            return self.app(environ, start_response)
-        except Exception, e:
-            if isinstance(e, PulpException):
-                status = e.http_status_code
-                response = serialization.error.http_error_obj(status, str(e))
-                response.update(e.data_dict())
-                response['error'] = e.to_dict()
-            else:
-                # If it's not a Pulp exception, return a 500
-                msg = _('Unhandled Exception')
-                logger.error(msg)
-                status = httplib.INTERNAL_SERVER_ERROR
-                response = serialization.error.http_error_obj(status, str(e))
-
-            if status == httplib.INTERNAL_SERVER_ERROR or self.debug:
-                logger.exception(str(e))
-                e_type, e_value, trace = sys.exc_info()
-                response['exception'] = traceback.format_exception_only(e_type, e_value)
-                response['traceback'] = traceback.format_tb(trace)
-            else:
-                logger.info(str(e))
-
-            serialized_response = json.dumps(response)
-            self.headers['Content-Length'] = str(len(serialized_response))
-            status_str = '%d %s' % (status, http_responses[status])
-            start_response(status_str, [(k, v) for k, v in self.headers.items()])
-            return [serialized_response]
-
-
 class DjangoExceptionHandlerMiddleware(object):
```

Comment:
Can this be renamed from `DjangoExceptionHandlerMiddleware` to `ExceptionHandlerMiddleware`. You'll have to make updates in other places to match.